kore/test/Test/Kore/Reachability/OnePathStrategy.hs

```diff
@@ -675,22 +674,23 @@ test_onePathStrategy =
                     , substitution = mempty
                     }
                 (fromTermLike Mock.a)
+            -- Is this right?
+            , Proven
```

Comment:
No, I don't think this is right.

Reply:
The `Proven` branch comes from simplifying a claim LHS which looks like this:
```
\and{testSort{}}(
    /* term: */
    /* Created: Test.Kore.Reachability.OnePathStrategy.functionalConstr10 */
    /* Fl Fn D Sfa Cl */
    functionalConstr10{}(/* Fl Fn D Sfa Cl */ b{}()),
\and{testSort{}}(
    /* predicate: */
    \and{testSort{}}(
        /* Sfa */
        \equals{boolSort{}, testSort{}}(
            /* Fn Sfa */
            lessIntId{}(
                /* Fn Sfa */ fTestInt{}(/* Fl Fn D Sfa Cl */ b{}()),
                /* Fl Fn D Sfa Cl */ /* builtin: */ \dv{intSort{}}("0")
            ),
            /* Fl Fn D Sfa Cl */ /* builtin: */ \dv{boolSort{}}("true")
        ),
    \and{testSort{}}(
        /* Sfa */
        \not{testSort{}}(
            /* Sfa */
            \equals{boolSort{}, testSort{}}(
                /* Fn Sfa */
                lessIntId{}(
                    /* Fn Sfa */ fTestInt{}(/* Fl Fn D Sfa Cl */ b{}()),
                    /* Fl Fn D Sfa Cl */ /* builtin: */ \dv{intSort{}}("0")
                ),
                /* Fl Fn D Sfa Cl */ /* builtin: */ \dv{boolSort{}}("false")
            )
        ),
    \and{testSort{}}(
        /* Sfa */
        \not{testSort{}}(
            /* Sfa */
            \equals{boolSort{}, testSort{}}(
                /* Fn Sfa */
                lessIntId{}(
                    /* Fn Sfa */ fTestInt{}(/* Fl Fn D Sfa Cl */ b{}()),
                    /* Fl Fn D Sfa Cl */ /* builtin: */ \dv{intSort{}}("0")
                ),
                /* Fl Fn D Sfa Cl */ /* builtin: */ \dv{boolSort{}}("true")
            )
        ),
        /* Spa */
        \ceil{testSort{}, testSort{}}(
            /* Created: Test.Kore.Reachability.OnePathStrategy.functionalConstr10 */
            /* Fl Fn D Sfa Cl */
            functionalConstr10{}(/* Fl Fn D Sfa Cl */ b{}())
        )
    ))),
    /* substitution: */
    \top{testSort{}}()
))
```
I'm trying to find out where this branch comes from.